src/main/java/org/elasticsearch/index/query/TermQueryBuilder.java

```diff
@@ -120,22 +147,36 @@ public TermQueryBuilder boost(float boost) {
     }
 
     /**
-     * Sets the query name for the filter that can be used when searching for matched_filters per hit.
+     * Gets the boost for this query.
+     */
+    public float boost() {
+        return this.boost;
+    }
+
+    /**
+     * Sets the query name for the query.
      */
     public TermQueryBuilder queryName(String queryName) {
         this.queryName = queryName;
         return this;
     }
 
+    /**
+     * Gets the query name for the query.
+     */
+    public String queryName() {
```

Comment:
at this point I 'd probably move the whole queryName code to the base class including serialization bits too. Up to you if you want to do it here or in a separate PR